### PR TITLE
feat: add document reader for GH issues

### PR DESCRIPTION
**Pull request by moncho**
The reader retrieves issues from a repo a converts them into documents. Documents text is the concatenation of the issue title and body, document Id is the issue number. Further information about the issue is added to the document extra_info field.

Issues can be filtered by their state (the default being open) and by label.

This is not part of the existing  GithubRepositoryReader because that loader works with repositories content, while this one works with repositories issues. As an alternative, the existing loader could be moved to a subdirectory under github_repo and this new loader could also be moved to under there, so there are separated but under the same root (github_repo), similar to `file` loaders. I guess this approach would break existing code, which is usually not good :)

### CODE REVIEW

File: llama_hub/github_repo_issues/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+https://github.com/jerryjliu/llama_index, use `owner = jerryjliu` and `repo = llama_index`.
+
+```shell
+export GITHUB_TOKEN='...'
+```
+
+```python
+import os
+
+from llama_hub.github_repo import GitHubRepositoryIssuesReader, GitHubIssuesClient
```
Comment (jerryjliu) [inline llama_hub/github_repo_issues/README.md:20]: github_repo_issues you mean?

File: llama_hub/github_repo_issues/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+```shell
+export OPENAI_API_KEY='...'
+export GITHUB_TOKEN='...'
+```
+
+```python
+import pickle
+import os
+
+from llama_index import download_loader, GPTVectorStoreIndex
```
Comment (jerryjliu) [inline llama_hub/github_repo_issues/README.md:50]: nit: it's just called a VectorStoreIndex now (GPT prefix is deprecated)

**Review by jerryjliu — CHANGES_REQUESTED** [review]
this looks great!! 

can you add an entry to libary.json? then it'll show up in the UI

File: llama_hub/github_repo_issues/README.md (hunk 1 above)
Comment (moncho) [reply llama_hub/github_repo_issues/README.md:20]: Yes, that is what I mean :)

Comment (moncho) [reply llama_hub/github_repo_issues/README.md]: Fixed!

File: llama_hub/github_repo_issues/README.md (hunk 2 above)
Comment (moncho) [reply llama_hub/github_repo_issues/README.md]: Changed

**Review by jerryjliu — APPROVED** [review]

File: llama_hub/library.json [hunk 3]
```diff
@@ -540,5 +540,15 @@
   "KibelaReader": {
     "id": "kibela",
     "author": "higebu"
+  },
+  "GitHubIssuesReader": {
```
Comment (jerryjliu) [inline llama_hub/library.json:544]: Sorry should've clarified, this needs to match the class name: GitHubRepositoryIssuesReader

Comment (moncho) [reply llama_hub/library.json]: Thanks for clarification, fixed!

**Pull request merged**